### PR TITLE
Add functionality to BAM module

### PR DESCRIPTION
**Pull request by andreadimaio**
Hi, I would like to share with you some work I did for the BAM module.

I added:
- Other parameters to the BamChatModel;
- Embedding and Tokenization APIs;
- Some tests

### CODE REVIEW

File: bam/runtime/src/main/java/io/quarkiverse/langchain4j/bam/runtime/BamRecorder.java [hunk 1]
```diff
@@ -18,23 +21,44 @@ public Supplier<?> chatModel(Langchain4jBamConfig runtimeConfig) {
...
+                .typicalP(firstOrDefault(null, chatModelConfig.typicalP()))
+                .repetitionPenalty(firstOrDefault(null, chatModelConfig.repetitionPenalty()))
+                .truncateInputTokens(firstOrDefault(null, chatModelConfig.truncateInputTokens()))
+                .beamWidth(firstOrDefault(null, chatModelConfig.beamWidth()));
+
+        if (runtimeConfig.baseUrl().isPresent()) {
+            builder.url(runtimeConfig.baseUrl().get());
+        }
+
+        return () -> builder.build();
```
Comment (geoand) [inline bam/runtime/src/main/java/io/quarkiverse/langchain4j/bam/runtime/BamRecorder.java:45]: Please avoid using lambdas in the `runtime` mave module.

File: bam/runtime/src/main/java/io/quarkiverse/langchain4j/bam/runtime/BamRecorder.java [hunk 2]
```diff
@@ -18,23 +21,44 @@ public Supplier<?> chatModel(Langchain4jBamConfig runtimeConfig) {
...
             builder.url(runtimeConfig.baseUrl().get());
         }
 
-        return new Supplier<>() {
-            @Override
-            public Object get() {
-                return builder.build();
-            }
-        };
+        return () -> builder.build();
```
Comment (geoand) [inline bam/runtime/src/main/java/io/quarkiverse/langchain4j/bam/runtime/BamRecorder.java:62]: Same here

**Review by geoand — COMMENTED** [review]
This looks really nice, thanks!

I added a small comment, while I also think the commit history can probably be cleaned uo a little

File: bam/runtime/src/main/java/io/quarkiverse/langchain4j/bam/runtime/BamRecorder.java (hunk 1 above)
Comment (andreadimaio) [reply bam/runtime/src/main/java/io/quarkiverse/langchain4j/bam/runtime/BamRecorder.java]: File updated without lambdas

File: bam/runtime/src/main/java/io/quarkiverse/langchain4j/bam/runtime/BamRecorder.java (hunk 2 above)
Comment (andreadimaio) [reply bam/runtime/src/main/java/io/quarkiverse/langchain4j/bam/runtime/BamRecorder.java]: File updated without lambdas

**Review by geoand — APPROVED** [review]
Very nice, thanks!

**Comment by geoand** [comment]
@andreadimaio do you mind addressing the CI issue? 

Once CI is green we can merge and release.

Thanks

**Comment by andreadimaio** [comment]
@geoand, yes I will fix it for the end of day. I have in mind also to commit some little changes to the code.

**Comment by geoand** [comment]
Cool, thanks

**Comment by andreadimaio** [comment]
@geoand you can find the last changes in the 54e7f57

**Comment by geoand** [comment]
I rebased and force pushed the PR in order to a new clean CI run

**Pull request merged**